server/application/application.go

```diff
@@ -207,8 +207,19 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*ap
 	if err != nil {
 		return nil, fmt.Errorf("error listing apps with selectors: %w", err)
 	}
+
+	filteredApps := apps
+	// Filter applications by name
+	filteredApps = argoutil.FilterByNameP(filteredApps, *q.Name)
```

Comment:
Probably need a nil check on q.Name. i.e. if it's nil, don't filter.

Reply:
eceb6d9
Sorry, corrected the issue as noted.